model/converter/json/from_domain.go

```diff
@@ -97,18 +97,26 @@ func (fd fromDomain) convertReferences(span *model.Span, preserveParentID bool)
 		length++
 	}
 	out := make([]json.Reference, 0, length)
-	if span.ParentSpanID != 0 && !preserveParentID {
-		out = append(out, json.Reference{
-			RefType: json.ChildOf,
-			TraceID: json.TraceID(span.TraceID.String()),
-			SpanID:  json.SpanID(span.ParentSpanID.String()),
-		})
-	}
+	var parentRefAdded bool
 	for _, ref := range span.References {
-		out = append(out, json.Reference{
+		newRef := json.Reference{
 			RefType: fd.convertRefType(ref.RefType),
 			TraceID: json.TraceID(ref.TraceID.String()),
 			SpanID:  json.SpanID(ref.SpanID.String()),
+		}
+		out = append(out, newRef)
+		if newRef.TraceID == json.TraceID(span.TraceID.String()) &&
```

Comment:
comparison via strings is inefficient - you have the strongly typed `ref` in scope, why not use that?